contentgrid-spring-data-rest/src/test/java/com/contentgrid/spring/data/querydsl/sort/CollectionFilterSortHandlerMethodArgumentResolverTest.java

```diff
@@ -0,0 +1,204 @@
+                                            pathNavigator.get("simpleString").getPath())
+                                    .lower()
+                    ));
+                });
+    }
+
+    @ParameterizedTest
+    @CsvSource({
+            "myInteger",
+            "myDecimal",
+            "myInstant"
+    })
+    void comparableSort(String param) {
+
+        var request = new MockHttpServletRequest("GET", "/entity-with-weird-filter-params");
+        request.addParameter("sort", param);
+        var sort = parseRequest(request);
+
+        var pathNavigator = new PathNavigator(entityPathResolver.createPath(EntityWithWeirdFilterParams.class));
+
+        assertThat(sort)
+                .containsExactly(Sort.Order.asc(param))
+                .isInstanceOfSatisfying(QSort.class, qSort -> {
+                    assertThat(qSort.getOrderSpecifiers()).containsExactly(new OrderSpecifier<>(
+                            Order.ASC,
+                            (Expression<Comparable<?>>) pathNavigator.get(param).getPath()
+                    ));
+                });
+    }
+
+    @Test
+    void unknownSortError() {
+        var request = new MockHttpServletRequest("GET", "/entity-with-weird-filter-params");
+        request.addParameter("sort", "xyz");
+        assertThatThrownBy(() -> parseRequest(request))
+                .hasMessage("Sort parameter 'xyz' is not supported")
+                .isInstanceOfSatisfying(UnsupportedSortPropertyException.class, ex -> {
+                    assertThat(ex.getOrder()).isEqualTo(Sort.Order.asc("xyz"));
+                });
+    }
+
+    @Configuration(proxyBeanMethods = false)
+    @EntityScan(basePackageClasses = {CollectionFilterSortHandlerMethodArgumentResolverTest.class,
+            InvoicingApplication.class})
+    @EnableJpaRepositories(basePackageClasses = {CollectionFilterSortHandlerMethodArgumentResolverTest.class,
+            InvoicingApplication.class}, considerNestedRepositories = true)
+    public static class LocalConfiguration {
+
+    }
+
+    @Entity
+    public static class EntityWithWeirdFilterParams {
+
+        @Id
+        @GeneratedValue(strategy = GenerationType.AUTO)
+        private UUID id;
+
+        @CollectionFilterParam(value = "simple_string", predicate = Text.EqualsNormalized.class)
+        @CollectionFilterParam(value = "simple_string_ignorecase", predicate = Text.EqualsIgnoreCaseNormalized.class)
+        @CollectionFilterParam(value = "simple_string_starts_with", predicate = Text.StartsWithIgnoreCaseNormalized.class)
+        private String simpleString;
+
+        @CollectionFilterParam
+        private Integer myInteger;
+
+        @CollectionFilterParam
+        private BigDecimal myDecimal;
+
+        @CollectionFilterParam
+        private Instant myInstant;
+
+        @OneToOne
+        @CollectionFilterParam(predicate = EntityId.class)
+        private com.contentgrid.spring.test.fixture.invoicing.model.Order order;
```

Comment:
Shouldn't there be a test that checks whether resolving a sort argument via a relation throws the `UnsupportedSortPropertyException`?